v2/internal/set/set.go

```diff
@@ -107,3 +107,14 @@ func AsSortedSlice[T constraints.Ordered](set Set[T]) []T {
 	slices.Sort(result)
 	return result
 }
+
+// Union returns the union of the specified sets
+func Union[T comparable](sets ...Set[T]) Set[T] {
+	result := make(Set[T])
```

Comment:
Should/could this be presized?

Reply:
Yeah it could be pre-sized. I initially didn't do that because it would be looping over the set of sets twice, but that's still probably more performant than doing extra allocations. I've updated to iterate the set of sets, get their length and preallocate.

Also added some tests.